### PR TITLE
Add default config to useConfig panel api

### PR DESCRIPTION
**Pull request by defunctzombie**
The default config is passed to useConfig and merges with any persisted
configuration.

### CODE REVIEW

File: app/PanelAPI/useConfig.test.tsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+//
+// This file incorporates work covered by the following copyright and
+// permission notice:
+//
+//   Copyright 2019-2021 Cruise LLC
+//
+//   This source code is licensed under the Apache License, Version 2.0,
+//   found at http://www.apache.org/licenses/LICENSE-2.0
+//   You may not use this file except in compliance with the License.
```
Comment (jtbandes) [inline app/PanelAPI/useConfig.test.tsx:17]: something is wrong here

File: app/PanelAPI/useConfig.test.tsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+          <PanelIdContext.Provider value="some-id">{children}</PanelIdContext.Provider>
+        </CurrentLayoutContext.Provider>
+      );
+    };
+
+    const { result } = renderHook(() => useConfig<{ foo: string }>({ foo: "bar" }), { wrapper });
+    expect(result.current[0]).toEqual({ foo: "bar" });
+
+    state.addPanelsStateListener((newState) => {
+      expect(newState.configById["some-id"]?.foo).toEqual("reset");
```
Comment (jtbandes) [inline app/PanelAPI/useConfig.test.tsx:106]: could also use a jest.fn with expect(f.mock.calls).toEqual, for more linear test flow (could then remove expect.assertions)

**Comment by jtbandes** [comment]
Has this been superseded by other changes at this point?

**Comment by defunctzombie** [comment]
Yea - This will need to be revisited in some other form because of how the panel settings sidebar wants to interact with panel settings. I'm not a fan of the current interactions and how we specify defaults today but this PR is stale and another approach will be needed.

**Pull request closed without merging**